chain/testing.go

```diff
@@ -65,12 +65,16 @@ func NewBuilder(t *testing.T, miner address.Address) *Builder {
 	b.messages[types.EmptyMessagesCID] = []*types.SignedMessage{}
 	b.receipts[types.EmptyReceiptsCID] = []*types.MessageReceipt{}
 
-	nullState, err := makeCid("null")
-	require.NoError(t, err)
+	nullState := types.CidFromString(t, "null")
 	b.tipStateCids[types.NewTipSetKey().String()] = nullState
 	return b
 }
 
+// SetStateBuilder sets the state builder
+func (f *Builder) SetStateBuilder(sb StateBuilder) {
+	f.stateBuilder = sb
+}
```

Comment:
Please replace this mutation with a new constructor, `NewBuilderWithState(...)` to which the existing constructor can delegate.